quick-start/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	vault "github.com/hashicorp/vault/api"
+)
+
+const (
+	vaultServerAddr = "http://127.0.0.1:8200"
+	token           = "dev-only-token" // This is the root token for our Vault dev server. Don't do this in production!
+	pathToSecret    = "secret/data/my-secret-password"
+	secretKey       = "password"
+	secretValue     = "Hashi123"
+)
+
+func main() {
+	config := vault.DefaultConfig()
+
+	config.Address = vaultServerAddr
+
+	client, err := vault.NewClient(config)
+	if err != nil {
+		log.Fatalf("unable to initialize Vault client: %v", err)
+	}
+
+	// Authentication
+	client.SetToken(token)
+
+	secretData := map[string]interface{}{
+		"data": map[string]interface{}{
+			secretKey: secretValue,
+		},
+	}
+
+	// Writing a secret
+	_, err = client.Logical().Write(pathToSecret, secretData)
+	if err != nil {
+		log.Fatalf("unable to write secret: %v", err)
+	}
+
+	fmt.Println("Secret written successfully.")
+
+	// Reading a secret
+	secret, err := client.Logical().Read(pathToSecret)
+	if err != nil {
+		log.Fatalf("unable to read secret: %v", err)
+	}
+
+	data, ok := secret.Data["data"].(map[string]interface{})
+	if !ok {
+		log.Fatalf("data type assertion failed: %T %#v", secret.Data["data"], secret.Data["data"])
+	}
+
+	value, ok := data[secretKey].(string)
+	if !ok {
+		log.Fatalf("value type assertion failed: %T %#v", data[secretKey], data[secretKey])
+	}
+
+	if value != secretValue {
+		log.Fatalf("unexpected password value %q retrieved from vault", value)
+	}
```

Comment:
If you decide to capture the response from `.Write()` you could compare `.Read()` to it and skip all of the checks here.

Reply:
I think it's still valuable to keep the checks here because typically the secret will only be read, not written, so people will mostly want to copy-pasta the read part.

Reply:
Agreed on keeping the checks. I think I'm gonna hold off on doing anything with the write response for now, since it kinda creates this "wait but if I have the secret object here, then why do I need to read it again on the subsequent lines...?" mental weirdness in my opinion.